apps/ledger-live-mobile/src/screens/SignMessage/Components/CopyButton.test.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { render, fireEvent } from "@tests/test-renderer";
+import { screen } from "@testing-library/react-native";
+import CopyButton from "./CopyButton";
+import Clipboard from "@react-native-clipboard/clipboard";
+
+jest.mock("@react-native-clipboard/clipboard", () => ({
+  setString: jest.fn().mockImplementation(() => Promise.resolve()),
+}));
+
+describe("CopyButton", () => {
+  const testText = "Text to copy";
+
+  beforeAll(() => {
+    Object.assign(navigator, {
+      clipboard: {
+        writeText: jest.fn().mockResolvedValue(undefined),
+      },
+    });
+  });
+
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("renders correctly with copy icon and text", () => {
+    render(<CopyButton text={testText} />);
+
+    expect(screen.getByTestId("copy-wrapper")).toBeVisible();
+    expect(screen.getByText("Copy")).toBeVisible();
+  });
+
+  it("copies text to clipboard when clicked", async () => {
+    render(<CopyButton text={testText} />);
```

Comment:
```suggestion
    const { user } = render(<CopyButton text={testText} />);
```
Then you can do await user.press(button) instead of fireEvent